datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -2869,7 +2869,11 @@ fn intersect_maps<'a>(
     let mut inputs = inputs.into_iter();
     let mut merged: HashMap<String, String> = inputs.next().cloned().unwrap_or_default();
     for input in inputs {
-        merged.retain(|k, v| input.get(k) == Some(v));
+        // The extra dereference below (`&*v`) is a workaround for https://github.com/rkyv/rkyv/issues/434.
+        // When this crate is used in a workspace that enables the `rkyv-64` feature in the `chrono` crate,
+        // this triggers a Rust compilation error:
+        // error[E0277]: can't compare `Option<&std::string::String>` with `Option<&mut std::string::String>`.
```

Comment:
I tried to reproduce the error, but didn't:
```rust
fn main() {
    let mut s =  String::from("hello");
    let ss = String::from("hello");
    let option_ref: Option<&String> = Some(&ss);
    let mut_ref = &mut s;

    if option_ref == Some(mut_ref) //Option<&mut String> {
    {
        println!("They are equal!");
    }
}
```
Maybe I didn't notice anything?

Reply:
is it compiler dependent? I was not able to reproduce either on latest compiler

Reply:
@xudong963 , please modify your example like this to reproduce:

* Add chrono as a dependency to Cargo.toml:

```toml
[dependencies]

chrono = { version = "0.4.38", default-features = false, features = ["rkyv-64"] }
```

* Add chrono import to main.rs:

```rust
use chrono::*;
```